### DISCUSSION TITLE
Create StatiqInfo(IDocument) Helper

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#206] [Ideas]
Inspired by `phpinfo()`, it would be useful to have a static helper class which accepts the current document and prints all document information (front matter, content), global settings, and other diagnostic information. This is helpful for developers who are building custom pages and are trying to identify what property needs to be retrieved or overridden in order to achieve the desired result.

Does anything like this exist currently or is this something we would be interested in adding to Statiq?

**Comment by daveaglick** [comment]
Yes! Well, sort of. Something like that sort of exists already, though I've done a terrible job of documenting it. Statiq has a full REPL you can use while in preview mode:

Running `Help()` at the top level to show what's available:
![image](https://github.com/statiqdev/Discussions/assets/1020407/76984988-d331-4904-8b95-713445201503)

And here's getting some specific metadata from a document:
![image](https://github.com/statiqdev/Discussions/assets/1020407/2a7204a8-207e-4a78-9235-12be7522fe4e)

That's not entirely the same as a full info method, but I love the idea - there's an existing issue to add something exactly like this to the REPL here: https://github.com/statiqdev/Statiq.Framework/issues/208
